packages/curvenote/src/session/session.ts

```diff
@@ -129,12 +122,32 @@ export class Session implements ISession {
     this._shownUpgrade = true;
   }
 
+  _clones: ISession[] = [];
+
   clone() {
-    return new Session(this.$tokens?.session ?? this.$tokens?.user, {
+    const cloneSession = new Session(this.$tokens?.session ?? this.$tokens?.user, {
       logger: this.log,
       apiUrl: this.API_URL,
       siteUrl: this.SITE_URL,
     });
+    this._clones.push(cloneSession);
+    return cloneSession;
+  }
+
+  getAllWarnings(ruleId: RuleId) {
```

Comment:
This is all an exact copy/paste from myst

Reply:
Looks good!!